### PR TITLE
Run environment variable validation before tests

### LINKED ISSUE #26: Running integration tests without a required variable causes them to fail without a clear error message
Error example

```
C:\devtools\idea2016.3.1\bin\runnerw.exe C:\tools\node\node.exe C:\tools\node\node_modules\npm\bin\npm-cli.js run test:integration

> koa-graphql-ts-template@0.1.0 test:integration C:\project\project-name
> jest --projects src/graphql

module.js:472
    throw err;
    ^

Error: Cannot find module 'C:\project\project-name\node_modules\jest-cli'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at getJest (C:\project\project-name\node_modules\jest\node_modules\jest-cli\build\cli\getJest.js:22:12)
    at Object.run (C:\project\project-name\node_modules\jest\node_modules\jest-cli\build\cli\index.js:39:48)
    at Object.<anonymous> (C:\project\project-name\node_modules\jest\node_modules\jest-cli\bin\jest.js:14:25)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! koa-graphql-ts-template@0.1.0 test:integration: `jest --projects src/graphql`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the koa-graphql-ts-template@0.1.0 test:integration script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\pavel.yeliseev\AppData\Roaming\npm-cache\_logs\2017-06-02T12_33_02_399Z-debug.log

Process finished with exit code 1
```

### PR DESCRIPTION
**Pull request by jurgisrudaks**
### Description
Because Jest has its own output rendering/buffering and `envalid` reporter sends an exit signal before Jest manages to render and output `console.errors` generated by reporter, if environment variables validation failed it just terminated with no error messages - i have made changes which will do environment variable validation before we even start Jest, which will bypass Jests buffering/rendering and output errors as expected.
### Relevant issue
Resolves #26

### CODE REVIEW

File: server.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import 'source-map-support/register';
-import env from './env';
+const env = require('./env');
```
Comment (trioletas) [inline server.ts]: Please avoid mixing `require`s and ES module `import`s. Suggest to use `import` everywhere for consistency with the front end codebases.

Comment (trioletas) [inline server.ts]: if you need to import `js` files, please consider `allowJs` [compiler option](https://www.typescriptlang.org/docs/handbook/compiler-options.html)

File: jest/__setup__/env-test.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 // read env to be accessible on every test (used in integration tests)
-import env from '../../env';
+const env = require('../../env');
```
Comment (trioletas) [inline jest/__setup__/env-test.ts]: same as for `server.ts`

File: package.json [hunk 3]
```diff
@@ -6,9 +6,9 @@
     "build": "npm run lint && tsc",
     "start": "npm run lint -- --force && run-p lint:watch build:watch",
     "test": "npm run test:all",
-    "test:integration": "jest --config jest/integration.config.js",
-    "test:unit": "jest --config jest/unit.config.js",
-    "test:all": "jest",
+    "test:integration": "node env.js && jest --config jest/integration.config.js",
```
Comment (trioletas) [inline package.json]: calling `node env.js` in three different tasks is kinda [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself)
let's extract to a standalone task for maintainability

**Comment by jurgisrudaks** [comment]
@trioletas thank you for the feedback - i totally agree, with it and i did a little refactor - now we use TypeScript and proper ES6 modules, as it already was - i just modified the validation task so it compiles the `env.ts` and then uses node to execute it - no more mixing of ES5 and 6 modules and JS and TypeScript.

**Comment by trioletas** [comment]
Cool, thanks!
I am testing this build on VSTS now and there are some new errors during the test phase that I don't quite understand yet.
It _could_ be related to the fact that we call raw 'node' executable from the path in 'validate:env' task.
Keep it unnmerged for now please.

**Comment by trioletas** [comment]
all good

**Pull request merged**